ui/src/app/zones/views/ZoneDetails/ZoneDetailsForm/ZoneDetailsForm.test.tsx

```diff
@@ -0,0 +1,81 @@
+import { mount } from "enzyme";
+import { act } from "react-dom/test-utils";
+import { Provider } from "react-redux";
+import configureStore from "redux-mock-store";
+
+import ZoneDetailsForm from "./ZoneDetailsForm";
+
+import type { RootState } from "app/store/root/types";
+import {
+  zone as zoneFactory,
+  zoneState as zoneStateFactory,
+  rootState as rootStateFactory,
+} from "testing/factories";
+
+const mockStore = configureStore();
+
+describe("ZoneDetailsForm", () => {
+  const testZone = {
+    desc: "test description",
+    id: 1,
+    name: "zone-name",
+  };
+  let initialState: RootState;
+
+  beforeEach(() => {
+    initialState = rootStateFactory({
+      zone: zoneStateFactory({
+        errors: {},
+        loading: false,
+        loaded: true,
+        items: [zoneFactory(testZone)],
```

Comment:
I'm surprised this works! The way the test factories work is they set a bunch of defaults, but you can override them if you want, e.g:
``` js
zoneFactory(); // returns all defaults
zoneFactory({ name: "different-name" }); // returns all defaults, except name is now "different-name"
```
But `desc` is not a valid property of a zone - `description` is. So what this is doing is it's adding an extra key for `desc` and keeping the default `description` value.

``` js
{
  id: 1,
  desc: 'test description',
  name: 'zone-name',
  controllers_count: 273270286,
  created: 'Wed, 08 Jul. 2020 05:35:4',
  description: 'test description',
  devices_count: 1197561872,
  machines_count: 1150783015,
  updated: 'Wed, 08 Jul. 2020 05:35:4'
}
```

TS correctly complains if you try to set `desc` directly in `zoneFactory`, but defining `testZone` first seems to be ok. I'm not sure why that is. :confused: